src/telemetry/sensor/mod.rs

```diff
@@ -36,27 +41,29 @@ impl Handle {
     where
         F: FnOnce() -> event::Event,
     {
-        if let Some(tx) = self.0.as_mut() {
-            // We may want to capture timestamps here instead of on the consumer-side...  That
-            // level of precision doesn't necessarily seem worth it yet.
-
+        if let Some(inner) = self.0.as_mut() {
             let ev = mk();
             trace!("event: {:?}", ev);
 
-            if tx.lossy_send(ev).is_err() {
-                debug!("dropped event");
+            if let Ok(mut taps) = inner.taps.lock() {
+                taps.inspect(&ev);
             }
+
+            inner.metrics.record_event(&ev);
```

Comment:
This also results in taking a mutex lock, right?

Reply:
yes, for now. I'd like to do away with the event model entirely so that the mutex is only needed when provisioning a route and not per-event or even per-request.